langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -308,13 +319,26 @@ private Map<String, Object> prefillMetadata() {
     private EmbeddingMatch<TextSegment> toEmbeddingMatch(Map<String, ?> item) {
         Map<String, ?> additional = (Map<String, ?>) item.get(ADDITIONALS);
         final Metadata metadata = new Metadata();
-        if (item.get(METADATA) != null && item.get(METADATA) instanceof Map) {
-            Map<String, ?> resultingMetadata = (Map<String, ?>) item.get(METADATA);
-            for (Map.Entry<String, ?> entry : resultingMetadata.entrySet()) {
-                if (entry.getValue() != null && !NULL_VALUE.equals(entry.getValue())) {
-                    metadata.add(entry.getKey(), entry.getValue());
-                }
+        Map<String, ?> metadataMap = new HashMap();
+        if (metadataFieldName == null || metadataFieldName.isEmpty()) {
+          metadataKeys.stream().forEach(key -> {
+              if(item.get(key) != null && !NULL_VALUE.equals(item.get(key))) {
+                // TODO: Remove or replace use of deprecated method
+                metadata.add(key, item.get(key));
+              }
             }
+          );
+        } 
+        else if (item.get(metadataFieldName) != null && item.get(metadataFieldName) instanceof Map) {
+          metadataMap = (Map<String, ?>) item.get(metadataFieldName);
+        }
+        if(metadataMap != null) {
```

Comment:
can `metadataMap` be null?

Reply:
No after looking at the code again I guess that is not possible

Reply:
Done